scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -288,10 +293,19 @@ private void createRemoteDirectoryIfNotExists(FTPClient ftpClient, String dirPat
     }
 
     private void createRemoteDirectory(FTPClient ftpClient, String remoteDirPath) throws IOException {
-        if (ftpClient.makeDirectory(remoteDirPath)) {
-            getOut().println("CREATED the directory: " + remoteDirPath);
+        ArrayList folderList = new ArrayList<>(Arrays.asList(ftpClient.listDirectories()));
+        ArrayList<String> folderNames = new ArrayList();
+        for (int i = 0; i < folderList.size(); i++) {
+            folderNames.add(((FTPFile) (folderList.get(i))).getName());
```

Comment:
You could remove the cast to `(FTPFile)` here if the generic of your ArrayList above was typed.